src/EventListener/InvalidIdListener.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ContentApiBundle\EventListener;
+
+use FluentDOM\DOM\Element;
+use Libero\ApiProblemBundle\Event\CreateApiProblem;
+use Libero\ContentApiBundle\Exception\InvalidId;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Translation\TranslatorInterface;
+
+final class InvalidIdListener
+{
+    private $translator;
+
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    public function onCreateApiProblem(CreateApiProblem $event) : void
+    {
+        /** @var Element $problem */
+        $problem = $event->getDocument()->documentElement;
+        $exception = $event->getException();
+
+        if (!$exception instanceof InvalidId) {
+            return;
+        }
+
+        $locale = $problem->getAttribute('xml:lang') ?? 'en';
+
+        $title = $this->translator->trans('libero.content.item.invalid_id.title', [], 'api_problem', $locale);
+        $details = $this->translator->trans(
+            'libero.content.item.invalid_id.details',
+            ['%id%' => $exception->getId()],
+            'api_problem',
+            $locale
+        );
+
+        $problem->appendElement('status', (string) Response::HTTP_BAD_REQUEST);
+        $problem->appendElement('title', $title);
+        $problem->appendElement('details', $details);
```

Comment:
These are assuming that they've not already been set. Should probably make it an upsert.